pkg/registry/common/interpose/server.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package interpose provides NetworkServiceRegistryServer that registers local Endpoints
+// and adds them to Map
+package interpose
+
+import (
+	"context"
+	"errors"
+	"net/url"
+	"strings"
+
+	"github.com/networkservicemesh/sdk/pkg/tools/interpose"
+
+	"github.com/google/uuid"
+
+	"github.com/networkservicemesh/sdk/pkg/registry/core/next"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/api/pkg/api/registry"
+)
+
+// InterposeNSEName - a common prefix for all registered cross NSEs
+const InterposeNSEName = "cross-connect-nse#"
```

Comment:
I think we can make this thing as a private const. For this, we need to add a registry/interpose client chain element that just checks incoming NSE name and if it has not this pattern then it should just append it. What do you think?

Reply:
This is a good Idea, I'd like it. Let me try. I think it should not check for name, but just replace it with Interpose required name.

Reply:
Fixed.